modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -636,9 +636,17 @@ public ObjectEntry deleteObjectEntry(ObjectEntry objectEntry)
 			_objectDefinitionPersistence.findByPrimaryKey(
 				objectEntry.getObjectDefinitionId());
 
-		_workflowInstanceLinkLocalService.deleteWorkflowInstanceLinks(
-			objectEntry.getCompanyId(), objectEntry.getNonzeroGroupId(),
-			objectDefinition.getClassName(), objectEntry.getObjectEntryId());
+		if (objectDefinition.isRootDescendantNode()) {
+			_startWorkflowInstance(
```

Comment:
Just to confirm if I understand correctly, we need to start the workflow instance from the root node because deleting the child is treated as an update in the tree?

Reply:
@guilhermedcamacho yes, that's correct. This is AC3:

![image](https://github.com/user-attachments/assets/77499fcd-9802-480c-bcc1-b94ec2ece065)